libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseFormFragment.java

```diff
@@ -359,7 +326,9 @@ public void onSiteChanged(SiteStore.OnSiteChanged event) {
             }
         }
 
-        startPostLoginServices();
+        if (mLoginListener instanceof LoginListener) {
```

Comment:
The reason was that we can't directly start the post-login services, because those belong to the `WordPress` app and not to the library. So I made that a generic action that's triggered through the `LoginListener`, but of course the `LoginBaseFormFragment` could be using a different type (e.g. the `LoginPrologueListener`).

I don't mind refactoring it that way - I also am not fond of making `LoginBaseFormFragment` aware of the `LoginListener` like that. However, most of the subclasses of the `LoginBaseFormFragment` will need to override it, so we're going to have a lot of instances of `mLoginListener.startPostLoginServices()`. I'll give it a go.

Reply:
Actually it looks like only `Login2FaFragment` needed this treatment - `LoginEmailPasswordFragment` and `LoginUsernamePasswordFragment` were already calling `startPostLoginServices` on their own terms.

Added the default method (overridden by `Login2FaFragment`) in fe094fc, though re-reading your comment you mentioned making that a `boolean` method. Was there a reason not to make it `void` that I'm missing?

Reply:
I had in mind a `boolean` function to just replace the `mLoginListener instanceof LoginListener` conditional, but the solution in c9d862e is even cleaner.

Come to thing of it though, what do you thing about letting `startPostLoginServices` happen at the client, inside `onLoginFinished()`? Currently, calling `onLoginFinished(true)` (notice the `true`) causes `onLoginFinished(no args)` to be called and that one is already overridden in the clients by design. Clients can then decide to call `startPostLoginServices` if they want to.

Reply:
This sounds good to me - I wanted to do something like this originally but tried to avoid making any unnecessary changes to the login logic. Since you also suggest this I'll work on making the change 👍

Reply:
I eliminated `LoginBaseFormFragment.startPostLoginServices()` in 8d189cd and moved the calls to `LoginListener.startPostLoginServices()` to `onLoginFinished()` in the client fragments that need it.

Reply:
Thanks for the extra work on this @aforcier ! I only left [a comment](https://github.com/wordpress-mobile/WordPress-Android/commit/8d189cd1e64b8efab32f0c3808745455f8dae8c8#r26715591) about the move from `LoginWpcomService` to the foreground UI.